py/defs.bzl

```diff
@@ -37,6 +38,32 @@ def _py_binary_or_test(name, rule, srcs, main, imports, deps = [], resolutions =
         tags = ["manual"],
     )
 
+def _find_main_impl(ctx):
+    found = None
+    for s in ctx.files.srcs:
+        if s.path.endswith(ctx.attr.main):
```

Comment:
This will keep searching over the sources after it has been found, either wastfully or potentially finding another match. What is the behaviour of upstream, does it match on the last or first? Perhaps we need a `break` in here.

Reply:
I went to locate the upstream behavior, it's already in starlark so it was easy to just vendor their `determine_main` logic.
(to answer your question, they also loop over all of them, then error if more than one match found)